### ISSUE TITLE
SQL text shema generation without database connection

### ISSUE THREAD

**Issue by GonistLelatel** [opis/database#62]
Idea: using the logic of the opis/database for generate the sql  text files.
Current code only allow makes it while the database is connected.
Create tables documentation: https://docs.opis.io/database/4.x/schema-creating-tables.html

**Comment by sorinsarca** [comment]
> Current code only allow makes it while the database is connected.

The current compiler needs an  `Opis\Database\Connection` because there are some commands that need extra info.
For example, in MySQL (< v8.0) to change only the column name you also have to specify the column type.

Changing `column_name` to `new_column_name`:

```sql
ALTER TABLE mytable CHANGE column_name new_column_name VARCHAR(100)
```

In order to get the current column type (that `VARCHAR(100)`) you have to ask the MySQL server,
and to ask the MySQL server you need a connection to it, so that's why you need that connection.

Starting with MySQL 8.0 you can use a new command which doesn't require the type:

```sql
ALTER TABLE mytable RENAME COLUMN column_name TO new_column_name
```

In this case a connection is not needed (at least for renaming a column).


In conclusion, since some drivers need extra info that can be provided only by the server, the compiler will require a connection.

Knowing this now, I have to ask: what's the issue?

**Comment by GonistLelatel** [comment]
this is an idea, not issue, and this idea has an example realization at xpdo_sql_shema package

**Comment by GonistLelatel** [comment]
Please see my realization in fork of this package.
https://github.com/GonistLelatel/opis_database/tree/master/example/schema
